Tasks/KubernetesManifestV0/src/utils/DeploymentHelper.ts

```diff
@@ -144,6 +160,50 @@ function updateImagePullSecretsInManifestFiles(filePaths: string[], imagePullSec
     return filePaths;
 }
 
+function captureAndPushDeploymentMetadata(filePaths: string[], allPods: any, deploymentStrategy: string, clusterInfo: any, manifestFilePaths: string[]) {
+    const requestUrl = getPublishDeploymentRequestUrl();
+    let metadata = {};
+    filePaths.forEach((filePath: string) => {
+        const fileContents = fs.readFileSync(filePath).toString();
+        yaml.safeLoadAll(fileContents, function (inputObject: any) {
+            if (!!inputObject && inputObject.kind && isDeploymentEntity(inputObject.kind)) {
+                metadata = getDeploymentMetadata(inputObject, allPods, deploymentStrategy, clusterInfo, manifestFilePaths);
+                pushDeploymentDataToEvidenceStore(JSON.stringify(metadata), requestUrl).then((result) => {
+                    tl.debug("DeploymentDetailsApiResponse: " + JSON.stringify(result));
+                }, (error) => {
+                    tl.warning("publishToImageMetadataStore failed with error: " + error);
+                });
+            }
+        });
+    });
+}
+
+async function pushDeploymentDataToEvidenceStore(requestBody: string, requestUrl: string): Promise<any> {
```

Comment:
This would have made more sense to be added to `kuberenetes-common` right? given that this snippet is duplicate

Reply:
This function has dependency on utility-common-v2/restutilities; And according to the current build rules, two util files under Common cannot have dependency on each other.

Reply:
You could move this to utility-common-v2 only then?